lbr_demos/lbr_demos_fri_ros2_cpp/src/cartesian_path_planning_node.cpp

```diff
@@ -0,0 +1,217 @@
+#include "rclcpp/rclcpp.hpp"
+#include "cmath"
+#include <iostream>
+#include <fstream>
+#include <sstream>
+#include <string>
+#include "chrono"
+#include "kdl_parser/kdl_parser.hpp"
+#include "kdl/chainfksolverpos_recursive.hpp" // for forward kinematics
+#include "kdl/chainiksolverpos_lma.hpp" // for inverse kinematics
+#include "lbr_fri_msgs/msg/lbr_position_command.hpp"
+#include "lbr_fri_msgs/msg/lbr_state.hpp"
+#include "friClientIf.h"
+
+
+using namespace std::chrono_literals; // this namespace to enable we can use 500ms to represent 500 milliseconds in our code.
+using std::placeholders::_1;
+
+lbr_fri_msgs::msg::LBRState current_robot_state; // store current robot state, including measured joint positions
+KDL::Frame initial_cartesian_position; // we want to move the end effector in sine curve in z-direction in Cartesian space, this is the initial position 
+
+class JointPositionPublisher:public rclcpp::Node
+{
+	private:
+		rclcpp::Publisher<lbr_fri_msgs::msg::LBRPositionCommand>::SharedPtr publisher_;
+        rclcpp::Subscription<lbr_fri_msgs::msg::LBRState>::SharedPtr subscriber_;
+
+        KDL::Chain chain_; // robot kinematics chain exetracted from robot URDF file
+
+        double amplitude_; // rad
+        double frequency_; // Hz
+        double sampling_time_; // sampling time for the state of robot and sending position command
+        double phase_; // initial phase
+		int count_;
+
+	private:
+        /**
+         * @function: callback function for Robot State Subscriber
+        */
+        void topic_callback(const lbr_fri_msgs::msg::LBRState& msg)
+        {
+            current_robot_state = msg;
+            
+            if(count_ < 200) // in first 2 seconds, to make 'initial_position_command' be 'measured_joint_position'
```

Comment:
might be better to introduce `is_init` bool or turn `current_robot_state` into unique pointer and check for null